include/sway/lock.h

```diff
@@ -0,0 +1,17 @@
+#ifndef _SWAY_LOCK_H
+#define _SWAY_LOCK_H
+#include <wlr/types/wlr_session_lock_v1.h>
+
+struct sway_session_lock_manager {
+	bool locked;
+
+	struct wlr_session_lock_v1 *lock;
+	struct wl_listener new_surface;
+	struct wl_listener unlock;
+	struct wl_listener abandon;
```

Comment:
To make it clear all of these events are triggered by `wlr_session_lock_v1`, maybe we can slap a `lock_` prefix? That way no need to introduce new wording like "abandon".